controllers/user.js

```diff
@@ -0,0 +1,33 @@
+let express = require("express"),
+    router = express.Router(),
+    userService = require("../services/user-service"),
+    emailService = require("../services/email-service");
+
+router.post('/', async function(req, res, next) {
+    try {
+        let patientEmail = req.body.patientEmail;
+        let patientLastName = req.body.patientLastName;
+        let patientFirstName = req.body.patientFirstName;
+        let parentEmail = req.body.parentEmail;
+        let parentLastName = req.body.parentLastName;
+        let parentFirstName = req.body.parentFirstName;
+
+        let patientUsername = userService.generateRandom();
+        let patientPassword = userService.generateRandom();
+        let parentUsername = userService.generateRandom();
+        let parentPassword = userService.generateRandom();
+
+        let patient = await userService.signUp(patientUsername, patientPassword, patientLastName, patientFirstName, patientEmail, "patient");

```

Comment:
we should consider putting this in a transaction and passing it to the user service. [This shows how to do this asyncronously](https://stackoverflow.com/questions/42870374/node-js-7-how-to-use-sequelize-transaction-with-async-await). The managed one looks fairly easy.